src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,243 @@
+                    if (input.length() == 5) {
+                        throw new InvalidArgumentsException();
+                    }
+                    String keyWord = input.substring(5);
+                    return new FindCommand(keyWord);
+                } catch (IndexOutOfBoundsException e) {
+                    throw new InvalidArgumentsException();
+                }
+            } else if (input.startsWith("todo")) {
+                return new TodoCommand((ToDo) parseTask(input, TaskType.TODO));
+            } else if (input.startsWith("deadline")) {
+                return new DeadlineCommand((Deadline) parseTask(input, TaskType.DEADLINE));
+            } else if (input.startsWith("event")) {
+                return new EventCommand((Event) parseTask(input, TaskType.EVENT));
+            } else {
+                throw new InvalidTaskException();
+            }
+        } catch (DukeException e) {
+            return new InvalidCommand(e);
+        }
+    }
+
+    /**
+     * Returns the Task parsed from file line.
+     *
+     * @param line File line to be parsed.
+     * @return Task parsed.
+     * @throws CorruptedFileException If file line is invalid.
+     */
+    public static Task parseFileLine(String line) throws CorruptedFileException {
+        String[] splitLine = line.split("%,", 4);
+        String input;
+        TaskType type;
+        boolean isDone = Boolean.parseBoolean(splitLine[1]);
+        switch (splitLine[0]) {
+        case "T":
+            input = "todo " + splitLine[2];
+            type = TaskType.TODO;
+            break;
+        case "D":
+            input = "deadline " + splitLine[2] + " /by " + splitLine[3];
+            type = TaskType.DEADLINE;
+            break;
+        case "E":
+            input = "event " + splitLine[2] + " /at " + splitLine[3];
+            type = TaskType.EVENT;
+            break;
+        default:
+            assert false : "Invalid task type";
+            throw new CorruptedFileException();
+        }
+        try {
+            Task task = parseTask(input, type);
+            if (isDone) {
+                task.markDone();
+            }
+            return task;
+        } catch (InvalidArgumentsException | InvalidTaskException e) {
+            throw new CorruptedFileException();
+        }
+    }
+
+    /**
+     * Returns the Task parsed from input.
+     *
+     * @param input Input to be parsed.
+     * @param type Type of Task.
+     * @return Task parsed.
+     * @throws InvalidArgumentsException If arguments are missing or invalid.
+     * @throws InvalidTaskException If input contains unknown commands.
+     */
+    private static Task parseTask(String input, TaskType type) throws InvalidArgumentsException, InvalidTaskException {
+        int descriptionEnd;
+        String description;
+        String dateTime;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+        Task task;
+        switch (type) {
```

Comment:
Good Job! It is very easy to read your code as the flow of the code is separated very logically and code is abstracted nicely into methods.